core/src/test/java/filemanagement/RandomImageFileMoverTest.java

```diff
@@ -265,5 +271,73 @@ public void restoreRandom_doMove() {
     when(randomNumberGenerator.nextInt(5)).thenReturn(0);
 
     assertTrue(randomImageFileMover.restoreRandom());
+
+    verify(fileWrappers[0]).move(moveDestinationFile);
+  }
+
+  @Test
+  public void restoreAll_emptyHiddenFolder() {
+    FileWrapper mockGalleryFolder = Mockito.mock(FileWrapper.class);
+    when(mockGalleryFolder.getFileName()).thenReturn("gallery");
+    when(mockGalleryFolder.getFilePath()).thenReturn("gallery");
+
+    FileWrapper mockHiddenFolder = Mockito.mock(FileWrapper.class);
+    when(mockHiddenFolder.getFileName()).thenReturn(".file-play");
+    when(mockHiddenFolder.getFilePath()).thenReturn("gallery/.file-play");
+    when(mockHiddenFolder.isDirectory()).thenReturn(true);
+
+    FileWrapper[] fileWrappers = {};
+
+    when(mockHiddenFolder.getFileList()).thenReturn(fileWrappers);
+
+    when(fileFactory.createFile(mockHiddenFolder.getFilePath())).thenReturn(mockHiddenFolder);
+    when(fileFactory.getGalleryFile()).thenReturn(mockGalleryFolder);
+
+    assertFalse(randomImageFileMover.restoreAll());
+  }
+

```

Comment:
Same as above, can you add some comments to clarify what the test for this function is doing?